src/shogun/features/CombinedDotFeatures.cpp

```diff
@@ -98,8 +103,8 @@ float64_t CCombinedDotFeatures::dot(int32_t vec_idx1, CDotFeatures* df, int32_t
 		ASSERT(f2)
 
 		result += f1->dot(vec_idx1, f2,vec_idx2) *
-			f1->get_combined_feature_weight() *
-			f2->get_combined_feature_weight();
+			get_subfeature_weight(f_idx) *
```

Comment:
this stuff can also be done in batch matrix operations if no subset is active...something for a future PR, as mentioned in #4643